spyder/api/plugins.py

```diff
@@ -89,25 +185,27 @@ def set_option(self, option, value, section=None):
           same or another plugin.
         * CONF_SECTION needs to be defined for this to work.
         """
-        super(BasePlugin, self)._set_option(option, value, section=section)
+        if self._config is not None:
+            section = self.CONF_SECTION if section is None else section
+            self._config.set(section, str(option), value)
 
-    def get_option(self, option, default=NoDefault, section=None):
+    @Slot(str)
+    @Slot(str, int)
+    def show_status_message(self, message, timeout=0):
         """
-        Get an option from Spyder configuration file.
+        Show message in main window's status bar.
 
         Parameters
         ----------
-        option: str
-            Name of the option to get its value from.
-
-        Returns
-        -------
-        bool, int, str, tuple, list, dict
-            Value associated with `option`.
+        message: str
+            Message to display in the status bar.
+        timeout: int
+            Amount of time to display the message.
         """
-        return super(BasePlugin, self)._get_option(option, default,
-                                                   section=section)
+        if self.main is not None:
+            self.main.statusBar().showMessage(message, timeout)
 
+    # FIXME: Should be start_
```

Comment:
This name makes sense because the plugin is not starting a long process here. It's only informing the main window that it's starting one.

Reply:
the `-ing`form of verbs feels odd in naming. Since this methods relate to actions, it makes more sense to use `on_`, `before_, `after_`

Reply:
Ok, `before_long_process` sounds better.